example/flaskr/cookbook.py

```diff
@@ -16,43 +16,91 @@
+
+@bp.route("/index", methods=("GET", "POST"))
+@login_required
+def index():
+    """Shows all recipes in a user's recipe book, POST method filters recipes shown, ordered most recent first."""
+    if g.user is None:
+        error = "Must Be a User to Have a Recipe Book"
+    
+    db = get_db()
+
+    sqlQuery = "SELECT * FROM recipe r JOIN recipeBook_recipe rbr ON \
+            rbr.recipeBook_id = (SELECT id FROM recipeBook WHERE author_id= ? ) \
+                JOIN recipe_cat rc ON r.category_id = rc.id "
+
     if request.method == "POST":
-        title = request.form["title"]
-        body = request.form["body"]
-        error = None
-
-        if not title:
-            error = "Title is required."
-        if g.user is None:
-            error = "Must Be a User to Save to Recipe Book"
-        if error is not None:
-            flash(error, "error")
-        else:
-            db = get_db()
-            db.execute(
-                "UPDATE recipe SET title = ?, body = ? WHERE id = ?", (title, body, id)
-            )
-            db.commit()
-            return redirect(url_for("recipe.index"))
-
-    return render_template("recipe/update.html", post=post, ingredients = ingredients)
-
-
-@bp.route("/<int:id>/delete", methods=("POST",))
+        sqlQuery += "WHERE " + \
+            request.form["filter"] + " LIKE '%" + \
+            request.form["search"] + "%' "
+
+        if request.form["category"] != 'NULL':
+            sqlQuery += "AND rc.name = '" + request.form["category"] + "' "
+
+    sqlQuery += "ORDER BY created DESC"
+
+    recipes = db.execute(sqlQuery, (g.user["id"],)).fetchall()
+    return render_template("cookbook/index.html", recipes=recipes)
+
+@bp.route("/add_recipe/<int:recipe_id>", methods=("GET",))
 @login_required
-def delete(id):
-    """Delete a recipe from cookbook.
+def save_recipe(recipe_id):
+    """Save recipe to current user's recipe book."""
+    error = None
 
-    Ensures that the recipe exists and that the logged in user is the
-    author of the recipe.
-    """
-    get_recipe(id)
+    if g.user is None:
+        error = "Must Be a User to Save to Recipe Book"
+    
     db = get_db()
-    db.execute("DELETE FROM recipe WHERE id = ?", (id,))
-    db.commit()
+    if (
+        db.execute("SELECT * FROM recipeBook_recipe WHERE \
+            recipeBook_id = (SELECT id FROM recipeBook WHERE author_id = ?)\
+                     AND recipe_id = ?",
+                   (g.user["id"],recipe_id)).fetchone()
+        is not None
+    ):
+        error = f"This recipe is already saved to your Recipe Book."
+
+    if error is not None:
+        flash(error, "error")
+    else:
+        get_recipe(recipe_id)
+        db.execute(
+            "INSERT INTO recipeBook_recipe (recipeBook_id, recipe_id) \
+            VALUES ( (SELECT id FROM recipeBook WHERE author_id = ?), ?)", 
+            (g.user["id"], recipe_id,)
+        )
+        db.commit()
+        flash(error, "Added to recipe book!")
```

Comment:
You can actually use the "info" category instead of "error" since this is supposed to be a positive message.